### PR TITLE
ART-1210: Auto run multiarch pre-release job for 4.3 and 4.4

### PR DESCRIPTION
**Pull request by vfreex**
Will run pre-prelease job in parallel for multiple versions and arches.

Test run with multiarch support:
https://saml.buildvm.openshift.eng.bos.redhat.com:8888/job/hack/job/yuxzhu/job/poll-payload/52/console

### CODE REVIEW

**Comment by vfreex** [comment]
Updated for multiarch.

File: scheduled-jobs/build/poll-payload/Jenkinsfile [hunk 1]
```diff
@@ -4,27 +4,37 @@
 import groovy.transform.Field
 import java.io.FileNotFoundException
 import java.net.URLEncoder
+import groovy.json.*
```
Comment (sosiouxme) [inline scheduled-jobs/build/poll-payload/Jenkinsfile]: bit broad, is there a need for anything beyond `JsonOutput`?

Comment (thiagoalessio) [reply scheduled-jobs/build/poll-payload/Jenkinsfile]: +1 let's import only what we need ;D

**Comment by sosiouxme** [comment]
aside from the conflict, this will need some rework for the separate arch-specific release-controllers.

**Comment by vfreex** [comment]
OK.

**Comment by vfreex** [comment]
Finished. Please do another review. Thanks.
@thiagoalessio @sosiouxme

**Comment by vfreex** [comment]
> @vfreex LGTM except I'm mildly worried about these complaints: https://saml.buildvm.openshift.eng.bos.redhat.com:8888/job/hack/job/yuxzhu/job/poll-payload/57/console
> 
> ```
> Invoking action for 4.3.0-0.nightly
> expected to call org.kohsuke.groovy.sandbox.impl.SandboxedMethodClosure.call but wound up catching WorkflowScript.startPreReleaseJob; see: https://jenkins.io/redirect/pipeline-cps-method-mismatches/
> [Pipeline] echo (hide)
> Invoking action for 4.4.0-0.nightly
> expected to call org.kohsuke.groovy.sandbox.impl.SandboxedMethodClosure.call but wound up catching WorkflowScript.startPreReleaseJob; see: https://jenkins.io/redirect/pipeline-cps-method-mismatches/
> ```
> 
> Not digging too deep but I think this is what happens when you use iterator methods; I'm not sure exactly what effect it has, other than apparently raising an error that makes the build come out UNSTABLE.

I noticed this warning during development. I believe it is a false positive warning when calling a method via a pointer, similar to https://issues.jenkins-ci.org/browse/JENKINS-59492. Those functions are called correctly without any issues.

The `UNSTABLE` result is not set by the warning. https://github.com/openshift/aos-cd-jobs/pull/2053/files#diff-b89f72b7074288d1245c7ece87537d33R99 and https://github.com/openshift/aos-cd-jobs/pull/2053/files#diff-b89f72b7074288d1245c7ece87537d33R114 set the build result to `UNSTABLE` when it fails to call the release controller API or downstream `pre-release` job fails. Currently many non-x86_64 API endpoints return 404, such as https://openshift-release-ppc64le.svc.ci.openshift.org/api/v1/releasestream/4.3.0-0.nightly-ppc64le/latest because no accepted payload exists.

**Comment by sosiouxme** [comment]
Thanks for the explanation. Hm, I'd like it to be OK to have missing endpoints (just ignore them) and report success but it doesn't seem like a reason to hold this up.

**Pull request merged**